packages/contracts-bedrock/scripts/libraries/DeployUtils.sol

```diff
@@ -217,12 +222,89 @@ library DeployUtils {
 
     /// @notice Asserts that the given proxy has an implementation set.
     /// @param _proxy Proxy to check.
-    function assertImplementationSet(address _proxy) internal {
+    function assertERC1967ImplementationSet(address _proxy) internal returns (address implementation_) {
         // We prank as the zero address due to the Proxy's `proxyCallIfNotAdmin` modifier.
         // Pranking inside this function also means it can no longer be considered `view`.
         vm.prank(address(0));
-        address implementation = IProxy(payable(_proxy)).implementation();
-        assertValidContractAddress(implementation);
+        implementation_ = IProxy(payable(_proxy)).implementation();
+        assertValidContractAddress(implementation_);
+    }
+
+    function assertL1ChugSplashImplementationSet(address _proxy) internal returns (address implementation_) {
+        vm.prank(address(0));
+        implementation_ = IStaticL1ChugSplashProxy(_proxy).getImplementation();
+        assertValidContractAddress(implementation_);
+    }
+
+    function assertResolvedDelegateProxyImplementationSet(
+        string memory _implementationName,
+        IAddressManager _addressManager
+    )
+        internal
+        view
+        returns (address implementation_)
+    {
+        implementation_ = _addressManager.getAddress(_implementationName);
+        assertValidContractAddress(implementation_);
+    }
+
+    function buildERC1967ProxyWithImpl(string memory proxyImplName) public returns (IProxy genericProxy) {
```

Comment:
Helper functions for building the three types of proxies with a dummy implementation (i.e. code at impl addr is `0x01`).